core/src/main/java/de/javakaffee/web/msm/MemcachedElasticConnectionFactory.java

```diff
@@ -0,0 +1,108 @@
+package de.javakaffee.web.msm;
+
+import net.spy.memcached.BinaryConnectionFactory;
+import net.spy.memcached.ConnectionFactory;
+import net.spy.memcached.DefaultConnectionFactory;
+
+import java.lang.reflect.*;
+
+/**
+ * Created by asodhi on 11/29/2016.
+ */
+public class MemcachedElasticConnectionFactory implements StorageClientFactory.MemcachedNodeURLBasedConnectionFactory {
+
+    @Override
+    public ConnectionFactory createBinaryConnectionFactory(final long operationTimeout,
+                                                           final long maxReconnectDelay,
+                                                           boolean isClientDynamicMode) {
+        try {
+            Class clientMode = Class.forName("net.spy.memcached.ClientMode");
+            Enum clientModeEnum = getClientMode(clientMode, isClientDynamicMode);
+            Class connectionFactoryClass = Class.forName("net.spy.memcached.BinaryConnectionFactory");
+            Constructor<BinaryConnectionFactory> connectionFactoryClassConstructor = connectionFactoryClass.getConstructor(clientMode);
+            BinaryConnectionFactory binaryConnectionFactory = connectionFactoryClassConstructor.newInstance(clientModeEnum);
+            return (ConnectionFactory) Proxy.newProxyInstance(
```

Comment:
trying to override the method  public long getOperationTimeout() in the BinaryConnectionFactory class.  No easy way to override methods when the class is created by reflection.